library/init/meta/tactic.lean

```diff
@@ -997,6 +997,20 @@ match _root_.try_for max (tac s) with
 | none   := mk_exception "try_for tactic failed, timeout" none s
 end
 
+meta def updateex_env (f : environment → exceptional environment) : tactic unit :=
+do env ← get_env,
+   env ← returnex $ f env,
+   set_env env
+
+meta def add_decl (d : declaration) : tactic unit :=
+updateex_env $ λe, do e ← e.add d, e.compile d tt
+
```

Comment:
What is the motivation for this change?
This seems to be a refactoring that has nothing to do with the coinductive command.
We need to discuss this kind of change. For example, I'm not sure I want to expose `compile` to users.